docs/readmes/proposals/p006_hook_terravm_to_ci.md

```diff
@@ -0,0 +1,247 @@
++-----------------------------------------------------+
+                          |
++-----------------------------------------------------+
+|                         |                           |
+|  AWS/Packet             | 2. Call API service       |
+|                         | with token                |
+|        +----------------v--------------------+      |
+|        | TipLab Bastion                      |      |
+|        | +----------+   +---------+  +-----+ |      |
+|        | | Api      |   |OpenVPN  |  | logs| |      |
+|        | | SERVICE  |   |SERVICE  |  |     | |      |
+|        | +----------+   +---------+  +-----+ |      |
+|        +-------------------------------------+      |
++-----------------------------------------------------+
+               |            |                |
++-----------------------------------------------------+
+|              |3.update    | 3.update       | 4.Trigger
+|  TipLab Infra|            |                | tests  |
+|              |            |                |        |
+|  +-----------v+  +--------v-----+   +------v------+ |
+|  |            |  |              |   |             | |
+|  |            |  |              |   |             | |
+|  |  AGW       |  |   FEG        |   |   NG40      | |
+|  |            |  |              |   |             | |
+|  |            |  |              |   |             | |
+|  +------------+  +--------------+   +-------------+ |
+|                                                     |
++-----------------------------------------------------+
+
+
+##### Components
+
+CircleCI Cloud:
+
+- CircleCI main component
+- CircleCI nodes : spinned up on the fly for every job
+
+AWS/Packet:
+
+- Tiplab Bastion: Access point to the tiplab infra   
+
+Tiplab Infra:
+
+- AGW
+- FEG
+- NG40: TeraVM "controller"
+
+##### Workflow
+
+1. CircleCi spins up a new CircleCI node that will have a token in order to
+call the api hosted on the TipLab Bastion
+
+2. CircleCI node call the Tiplab bastion with the token and trigger a test
+execution
+
+3. Tiplab bastion can access the TipLab Infra and will update both agw and feg
+with the latest available tag using the FabFile (in the future those packages
+can be build directly on the CI node and transferred to the components directly
+or uploaded to a special TeraVM registry)
+
+3. Tiplab bastion can access the TipLab Infra and will use the fabfile to
+Trigger ng40 tests.
+
+##### Pros
+
+- The Safest option
+- We control the infra that run the VPN service
+- Another layer of authentification using tokens
+- We can dig in the option of whitelisting the CI nodes (Not sure if it's
+possible)
+- TipLab Bastion is reusable for other specific infra
+- More Control on logs/access
+
+##### Cons
+
+- New components are thrown in the picture (Tiplab Bastion)
+- More dev time
+- More maintenance
+
+
+#### Design 3
+
+Using TeraVM cloud (AWS). No design for this option as I don't know the
+feasibility of it.
```

Comment:
in this option we should factor in the cost of AWS. Teravm would use lots of cpu cycles so not sure if it is worth it the cost when we already have the infra in place

Reply:
We also need information about who is hosting it etc .. that's why I left it "blank"